public/app/features/provisioning/Wizard/actions.ts

```diff
@@ -0,0 +1,133 @@
+import { GetRepositoryFilesResponse, GetResourceStatsResponse, RepositoryViewList } from '../api';
+
+export type Target = 'instance' | 'folder';
+export type Operation = 'pull' | 'migrate';
+
+export interface ModeOption {
+  target: Target;
+  operation: Operation;
+  label: string;
+  description: string;
+  disabledReason?: string;
+}
+
+export interface SystemState {
+  loading: boolean;
+  resourceCount: number;
+  fileCount: number;
+  actions: ModeOption[];
+  disabled: ModeOption[];
+  folderConnected?: boolean;
+}
+
+const migrateInstance: ModeOption = {
+  target: 'instance',
+  operation: 'migrate',
+  label: 'Migrate Instance to Repository',
+  description: 'Save all Grafana resources to repository',
+};
+
+const pullInstance: ModeOption = {
+  target: 'instance',
+  operation: 'pull',
+  label: 'Pull from Repository to Instance',
+  description: 'Pull resources from repository into this Grafana instance',
+};
+const pullFolder: ModeOption = {
+  target: 'folder',
+  operation: 'pull',
+  label: 'Pull from Repository to Folder',
+  description: 'Pull repository resources into a specific folder',
+};
+
+export function getState(
+  loading: boolean,
+  repoName: string,
+  settings?: RepositoryViewList,
+  files?: GetRepositoryFilesResponse,
+  stats?: GetResourceStatsResponse
+): SystemState {
+  const state: SystemState = {
+    loading,
+    resourceCount: 0,
+    fileCount: 0,
+    actions: [],
+    disabled: [],
+  };
+  if (settings?.items) {
+    settings.items.forEach((v) => {
+      if (v.name === repoName) {
+        return; // do not count yourself
+      }
+      if (v.target === 'folder') {
+        state.folderConnected = true;
+      }
+      if (v.target === 'instance') {
+        state.folderConnected = true;
+      }
+    });
+  }
+
+  if (files?.items) {
+    files.items.forEach((v) => {
+      const p = v.path ?? '';
+      if (p.endsWith('.json') || p.endsWith('.yaml')) {
```

Comment:
🤔 we should perhaps *only* include .json + yaml

Reply:
yes, I was thinking about this